src/services/user.service.test.js

```diff
@@ -1,57 +1,227 @@
+  register, login, me, refresh,
+} = require('./user.service');
 
 jest.mock('bcryptjs', () => ({
   genSalt: jest.fn(),
   hash: jest.fn(),
+  compare: jest.fn(),
+}));
+
+jest.mock('jsonwebtoken', () => ({
+  verify: jest.fn(),
+  sign: jest.fn().mockReturnValue('mocked-token'),
 }));
 
 jest.mock('../models/user.model', () => jest.fn().mockImplementation(() => ({
   save: jest.fn(),
 })));
 
-describe('add function', () => {
-  afterEach(() => {
+describe('Function register', () => {
+  beforeEach(() => {
     jest.clearAllMocks();
+    process.env.ACCESS_TOKEN_TTL = '2h';
   });
 
-  it('should hash password and save user', async () => {
-    const username = 'testuser';
-    const password = 'testpassword';
-    const mockedSalt = 'mockedSalt';
-    const mockedHashedPassword = 'mockedHashedPassword';
-    const mockedUuid = 'mockedUuid';
-
-    bcrypt.genSalt.mockResolvedValue(mockedSalt);
-    bcrypt.hash.mockResolvedValue(mockedHashedPassword);
+  it('registers a new user successfully', async () => {
+    const mockUsername = 'newuser';
+    const mockPassword = 'testpassword';
 
     User.mockImplementation(() => ({
       save: jest.fn().mockResolvedValue({
-        username,
-        password: mockedHashedPassword,
-        uuid: mockedUuid,
+        username: mockUsername,
+        password: 'mocked-hash',
+        uuid: expect.any(String),
         playedGames: [],
       }),
     }));
+    User.findOne = jest.fn().mockResolvedValue(null);
+    bcrypt.genSalt = jest.fn().mockResolvedValue('mocked-salt');
+    bcrypt.hash = jest.fn().mockResolvedValue('mocked-hash');
 
-    const result = await add(username, password);
+    const result = await register(mockUsername, mockPassword);
 
+    expect(User.findOne).toHaveBeenCalledWith({ username: mockUsername });
     expect(bcrypt.genSalt).toHaveBeenCalledWith(10);
-    expect(bcrypt.hash).toHaveBeenCalledWith(password, mockedSalt);
+    expect(bcrypt.hash).toHaveBeenCalledWith(mockPassword, 'mocked-salt');
 
-    expect(User).toHaveBeenCalledWith({
-      username,
-      password: mockedHashedPassword,
-      uuid: expect.any(String),
-      playedGames: [],
+    expect(result).toEqual({
+      accessToken: 'mocked-token',
+      refreshToken: 'mocked-token',
+      expires_in: process.env.ACCESS_TOKEN_TTL,
+    });
+  });
+
+  it('rejects registration if error while getting user', async () => {
+    const mockUsername = 'existinguser';
+    const mockPassword = 'testpassword';
+
+    // Mocking getByUsername to return an existing user
+    User.findOne = jest.fn().mockImplementation(() => {
+      throw new Error('err');
     });
 
+    await expect(register(mockUsername, mockPassword)).rejects.toThrow('err');
+  });
+
+  it('rejects registration if user already exists', async () => {
+    const mockUsername = 'existinguser';
+    const mockPassword = 'testpassword';
+
+    // Mocking getByUsername to return an existing user
+    User.findOne = jest.fn().mockResolvedValue({});
```

Comment:
If the user exists, it should probably find a non-empty value.

Reply:
For this it is not needed to set a correct user in the response since the only relevant part is that it does not return null

Reply:
Functionality-wise, it doesn't make a difference, but it would make the test more straightforward to understand if we return a correct user object. But the change is optional for now